lib/streams.js

```diff
@@ -54,3 +54,79 @@ class WriteStream extends Writable {
 }
 
 exports.WriteStream = WriteStream
+
+class ByteStream extends Readable {
+  constructor (core, opts = {}) {
+    super()
+
+    this.core = core
+    this.byteOffset = opts.byteOffset || 0
+    this.byteLength = typeof opts.byteLength === 'number' ? opts.byteLength : -1
+    this.prefetch = typeof opts.prefetch === 'number' ? opts.prefetch : 32
+
+    this.index = 0
+    this.range = null
+  }
+
+  _open (cb) {
+    this._openp().then(cb, cb)
+  }
+
+  _read (cb) {
+    this._readp().then(cb, cb)
+  }
+
+  async _openp () {
+    if (this.byteLength === -1) {
+      await this.core.update()
+      this.byteLength = Math.max(this.core.byteLength - this.byteOffset, 0)
+    }
+  }
+
+  async _readp () {
+    let data = null
+
+    if (this.byteLength === 0) {
+      this.push(null)
+      return
+    }
+
+    let relativeOffset = 0
+
+    if (this.byteOffset > 0) {
+      const [block, byteOffset] = await this.core.seek(this.byteOffset)
+
+      this.byteOffset = 0
```

Comment:
dont you think this should be _byteOffset since we set it to 0?

Reply:
True, I added a boolean to avoid mutating it but still `byteLength` needs to be private so not cool to just have one prop